config/config.go

```diff
@@ -511,8 +544,25 @@ func (c *Config) LoadAll(configFiles ...string) error {
 	return c.LinkSecrets()
 }
 
+type cfgDataOptions struct {
+	sourcePath string
+}
+
+type cfgDataOption func(*cfgDataOptions)
+
+func WithSourcePath(path string) cfgDataOption {
+	return func(o *cfgDataOptions) {
+		o.sourcePath = path
+	}
+}
+
 // LoadConfigData loads TOML-formatted config data
-func (c *Config) LoadConfigData(data []byte) error {
+func (c *Config) LoadConfigData(data []byte, opts ...cfgDataOption) error {
```

Comment:
I think you should just pass the `source` here. If we ever want to add other flags we can still refactor this later.

Reply:
I see your point but my fear is that people might forget about adding the source in future changes and we will not notice. You might keep this as is but I really would feel better if we enforce providing a source... ;-)

Reply:
I am inclined to keep this as-is for this PR. I can open up a second one to enforce the func signature change. Let me know your thoughts.